File: searches/koko_eating_bananas.rb
def min_eating_speed(piles, h)
    left = 1
    right = piles.max

    puts piles

    def valid_speed(piles, spd) 
        piles.sum{ |num| (num.to_f / spd).ceil }
	  end

    while left < right
        mid = (left + right) / 2
        puts mid
        if valid_speed(piles, mid) > h
            left = mid + 1
        else
            right = mid
        end	
    end
    
    left
end

File: searches/koko_eating_bananas.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        left = 1
        right = max(piles)

        def valid_speed(piles, spd):
            return sum( math.ceil(num / spd) for num in piles )

        while left < right:
            mid = (left + right) // 2
            if valid_speed(piles, mid) > h:
                left = mid + 1
            else:
                right = mid
    
        return left